### PR TITLE
Updating to the latest upstream '7.0.0-rc-4.0' version

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Updating to the latest upstream '7.0.0-rc-4.0' version

### What issues does this PR fix or reference?
https://github.com/redhat-developer/rh-che/issues/1489

### How have you tested this PR?
PR check CI

### CODE REVIEW

**Comment by amisevsk** [comment]
From the `check-upstream-dashboard.sh` script:
```
Upstream changes detected. Changed files:

./assembly/fabric8-ide-dashboard-war/target/sources/src/app/workspaces/create-workspace/create-workspace.controller.ts
./assembly/fabric8-ide-dashboard-war/target/sources/src/app/workspaces/create-workspace/create-workspace.html
./assembly/fabric8-ide-dashboard-war/target/sources/src/components/widget/footer/che-footer.html
```
But I haven't looked into it further.

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
Currently failing with:

> [INFO] Compiling 19 source files to /root/payload/plugins/fabric8-multi-tenant-manager/target/classes
> /root/payload/plugins/fabric8-multi-tenant-manager/src/main/java/com/redhat/che/multitenant/RhCheInfraEnvironmentProvisioner.java:94: error: constructor OpenShiftEnvironmentProvisioner in class OpenShiftEnvironmentProvisioner cannot be applied to given types;
>     super(
>     ^
>   required: boolean,OpenShiftUniqueNamesProvisioner,RouteTlsProvisioner,ServersConverter<OpenShiftEnvironment>,EnvVarsConverter,RestartPolicyRewriter,WorkspaceVolumesStrategy,RamLimitRequestProvisioner,InstallerServersPortProvisioner,LogsVolumeMachineProvisioner,PodTerminationGracePeriodProvisioner,ImagePullSecretProvisioner,ProxySettingsProvisioner,ServiceAccountProvisioner,CertificateProvisioner,VcsSshKeysProvisioner
>   found: boolean,OpenShiftUniqueNamesProvisioner,RouteTlsProvisioner,ServersConverter<OpenShiftEnvironment>,EnvVarsConverter,RestartPolicyRewriter,WorkspaceVolumesStrategy,RamLimitRequestProvisioner,InstallerServersPortProvisioner,LogsVolumeMachineProvisioner,PodTerminationGracePeriodProvisioner,ImagePullSecretProvisioner,ProxySettingsProvisioner,ServiceAccountProvisioner,CertificateProvisioner
>   reason: actual and formal argument lists differ in length

**Comment by amisevsk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by amisevsk** [comment]
Failing on functional tests, which isn't surprising since the dashboard has changed in `rc-4.0`.

**Comment by ibuziuk** [comment]
@ScrewTSW @Katka92 could you please investigate the test failures - https://ci.centos.org/job/devtools-rh-che-rh-che-prcheck-dev.rdu2c.fabric8.io/1685/console

**Comment by ibuziuk** [comment]
PR is ready for review. Once test are fixed we will be able to promote `7.0.0-rc-4.0` to prod-preview

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
[test]

**Comment by ibuziuk** [comment]
Ok, looks like we need to update the build due to the initial cleanup of Che 6 legacy code - https://github.com/eclipse/che/pull/13788

**Comment by amisevsk** [comment]
[test]

**Comment by ibuziuk** [comment]
@amisevsk I reverted back changes for build fix and push it for now to  https://github.com/ibuziuk/rh-che/commits/rc4-fix (even though the PR was buildable rh-che pod startup failed with `cp: cannot stat ‘/home/user/eclipse-che//templates/*’: No such file or directory`)

**Comment by amisevsk** [comment]
@ScrewTSW @Katka92 PTAL at the e2e changes before we merge.

File: assembly/assembly-ide-war/pom.xml [hunk 1]
```diff
@@ -48,7 +41,7 @@
         </dependency>
         <dependency>
             <groupId>org.eclipse.che</groupId>
-            <artifactId>assembly-ide-war</artifactId>
+            <artifactId>assembly-root-war</artifactId>
```
Comment (ibuziuk) [inline assembly/assembly-ide-war/pom.xml]: maybe it makes sense to rename the folder also to `assembly-root-war` to make it consistent ?

Comment (ibuziuk) [reply assembly/assembly-ide-war/pom.xml]: we could probably consider renaming as part of another PR

Comment (amisevsk) [reply assembly/assembly-ide-war/pom.xml]: I'm +1 on renaming, I didn't do it here to avoid messing with the build too much at once.

Comment (ibuziuk) [reply assembly/assembly-ide-war/pom.xml]: agree PR was already BIG, we could handle it separately

**Pull request merged**